packages/network-console-shared/src/file_io/convert.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { INetConsoleAuthorization } from '../net/net-console-http';
+import {
+    ICollectionFormat,
+    ICollectionAdapter,
+    ICollectionContainerAdapter,
+    ICollectionItemAdapter,
+} from './interfaces';
+
+export async function convertFormats(source: ICollectionAdapter, target: ICollectionFormat): Promise<ICollectionAdapter> {
+    if (!target.canWrite) {
+        throw new RangeError('Destination format is not writable.');
+    }
+
+    const targetCollection = await target.createCollection(source.name);
+    for (const id of source.childEntryIds) {
+        const child = source.getEntryById(id);
+        if (child!.type === 'container') {
+            await convertContainer(child as ICollectionContainerAdapter, targetCollection);
+        }
+        else {
+            await convertRequest(child as ICollectionItemAdapter, targetCollection);
+        }

```

Comment:
I think this can perform better if you add them to an array and execute them with Promise.all. It depends on what they end up doing in each method, but from my little understanding on the code it may help.

Reply:
Since we're not streaming data from disk, it really doesn't matter. The entirety of the collection is already loaded into memory, so we'll just be CPU-bound.